Frameworks/MQTTnet.NetStandard/Server/MqttClientPendingMessagesQueue.cs

```diff
@@ -12,7 +12,8 @@ namespace MQTTnet.Server
 {
     public sealed class MqttClientPendingMessagesQueue : IDisposable
     {
-        private readonly BlockingCollection<MqttBasePacket> _queue = new BlockingCollection<MqttBasePacket>();
+        private readonly ConcurrentQueue<MqttBasePacket> _queue = new ConcurrentQueue<MqttBasePacket>();
+        private readonly SemaphoreSlim _queueWaitSemaphore = new SemaphoreSlim(0);
```

Comment:
Is 0 correct here? I also used the Semaphore for several places but I had to use a different overload.

Reply:
According to the docs, `new SemaphoreSlim(0)` means the initial count of the SemaphoreSlim is 0, but there is no maximum count. Internally, .NET Framework uses `int32.MaxValue` for this.

The semaphore in this case represents the number of packets that are currently in the queue, which can be more than 1, so I think this should be correct.